plugins/content/caption/base/services/CaptionAssetService.php

```diff
@@ -201,7 +201,13 @@ protected function attachFile(CaptionAsset $captionAsset, $fullPath, $copyOnly =
 
 		$finalPath = kFileSyncUtils::getLocalFilePathForKey($syncKey);
 		list($width, $height, $type, $attr) = getimagesize($finalPath);
-		
+
+		if ($this->detectMultiLanguageFile($captionAsset , $finalPath))

```

Comment:
No reason to parse the content to detect it, just set it to multilingual when you ingest it.